File: LCOF/9.CQueue.cpp
class CQueue {
public:
    stack<int> q;
    stack<int> qq;

    CQueue() {

    }
    
    void appendTail(int value) {
        if (q.empty()) {
            while (!qq.empty()) {
                int val = qq.top();
                qq.pop();
                q.push(val);
            }
        }
        q.push(value);
    }
    
    int deleteHead() {
        while(!q.empty()) {
            int val = q.top();
            q.pop();
            qq.push(val);
        }
        if (qq.empty()) return -1;
        int val = qq.top();
        qq.pop();
        
        return val;        
    }
};

/**
 * Your CQueue object will be instantiated and called as such:
 * CQueue* obj = new CQueue();
 * obj->appendTail(value);
 * int param_2 = obj->deleteHead();
 */

File: LCOF/9.CQueue.go
type CQueue struct {
    stack1, stack2 *list.List
}


func Constructor() CQueue {
    return CQueue {
        stack1: list.New(),
        stack2: list.New(),
    }
}


func (this *CQueue) AppendTail(value int)  {
    this.stack1.PushBack(value)
}


func (this *CQueue) DeleteHead() int {
    if (this.stack2.Len() == 0) {
        for (this.stack1.Len() != 0) {
            e := this.stack1.Remove(this.stack1.Back())
            this.stack2.PushBack(e.(int))
        }
    }
    if (this.stack2.Len() != 0) {
        e := this.stack2.Remove(this.stack2.Back())
        return e.(int)
    }

    return -1
}


/**
 * Your CQueue object will be instantiated and called as such:
 * obj := Constructor();
 * obj.AppendTail(value);
 * param_2 := obj.DeleteHead();
 */